src/EditorFeatures/CSharp/MisplacedUsingDirectives/MisplacedUsingDirectivesCodeFixProvider.cs

```diff
@@ -0,0 +1,394 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.AddImports;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Simplification;
+using Roslyn.Utilities;
+using static Microsoft.CodeAnalysis.CodeActions.CodeAction;
+
+namespace Microsoft.CodeAnalysis.CSharp.MisplacedUsingDirectives
+{
+    /// <summary>
+    /// Implements a code fix for all misplaced using statements.
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.MoveMisplacedUsings)]
+    [Shared]
+    internal sealed partial class MisplacedUsingDirectivesCodeFixProvider : CodeFixProvider
+    {
+        private static readonly SyntaxAnnotation s_usingPlacementCodeFixAnnotation = new SyntaxAnnotation(nameof(s_usingPlacementCodeFixAnnotation));
+
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(IDEDiagnosticIds.MoveMisplacedUsingsDiagnosticId);
+
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return FixAll.Instance;
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var syntaxRoot = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var compilationUnit = (CompilationUnitSyntax)syntaxRoot;
+            var options = await context.Document.GetOptionsAsync(context.CancellationToken).ConfigureAwait(false);
+
+            // Do not offer a code fix when there are multiple namespaces in the source file.
+            if (CountNamespaces(compilationUnit.Members) > 1)
+            {
+                return;
+            }
```

Comment:
is it 'multiple namespaces'.  or is 'more than one member'.  For example, if i have this:

```c#
using System;

namespace Foo {
   ...
}

class C { 
}
```

We can't move teh using safely into the namespcae, because 'C' might break.

Reply:
The fixer does not offer a fix when there are multiple members. So I guess it comes down to whether there is a value in reporting a diagnostic that doesn't have a fix.

Reply:
i see.  this is worth commenting in the code.  it wasn't clear to me immediately the relation between the analyzer and the fixer and which checks each does (and why).